### ISSUE TITLE
about deployment

### ISSUE THREAD

**Issue by Indiasuperman** [XanderWangXY/Deployment-for-Aliengo#3]
大佬您的parkour成功了吗？我现在也在尝试部署您的开源代码，想问问您当初run.py是怎么运行的？是在自己pc上运行的吗？如何连接的机器狗？我对机器狗目前知道甚少，希望得到您的帮助。非常感谢开源

**Comment by XanderWangXY** [comment]
我这个代码没法用，就是自己学习时候随便写的。你可以看看这个项目，有部署的代码：https://github.com/ZiwenZhuang/parkour
如果不用摄像头的话可以运行在其他主机上，用网线连上自己的主机和狗，然后改一下自己主机网口的ip就行了，具体的看这个：https://www.yuque.com/ironfatty/nly1un
用摄像头的话可能要自己想办法把图像传到电脑上，比较麻烦吧，还是运行在狗上比较合适。狗上有两块嵌入式计算机，一块arm架构的jetson tx2和一块x86架构的迷你pc，代码应该运行在tx2上，所以当你配置环境遇到问题时也可以上网查查jetson系列产品的资料

**Comment by Indiasuperman** [comment]
我也一直在尝试部署extreme-parkour项目，但是源代码中的observations输入一直没搞清楚，比如num_scan这些

**Comment by XanderWangXY** [comment]
的确，observation太多了，好多东西不知道是啥

**Comment by Indiasuperman** [comment]
大佬的整体框架没问题我感觉，我准备在这个基础上改observations输入，估计就是输入问题

**Comment by heartInsert** [comment]
> 大佬的整体框架没问题我感觉，我准备在这个基础上改observations输入，估计就是输入问题
你的observation 可以了吗，线性速度这个值你有得到吗？

**Comment by XanderWangXY** [comment]
> > 大佬的整体框架没问题我感觉，我准备在这个基础上改observations输入，估计就是输入问题
> > 你的observation 可以了吗，线性速度这个值你有得到吗？
没有，我那个线速度算的有问题。

**Comment by Indiasuperman** [comment]
哪个线性速度，我现在配环境配癫了快

**Comment by Indiasuperman** [comment]
self.observation = torch.cat((self.obs_buf,
                                      self.get_useless(),
                                      #self.get_heights(),
                                      #self.get_priv_explicit(),
                                      #self.get_priv_latent(),
                                      self.obs_history_buf.view(1, -1)
                                      ), dim=-1) 那些privileged info 全变0就好了，170维都是无用的信息

**Comment by Indiasuperman** [comment]
@heartInsert 老哥你解决宇树sdk问题了吗，我python3.6在编译成功并且生成两个arm64下的.so文件后，还是显示no module named robot_interface

**Comment by heartInsert** [comment]
> @heartInsert 老哥你解决宇树sdk问题了吗，我python3.6在编译成功并且生成两个arm64下的.so文件后，还是显示no module named robot_interface

SDK我解决了，我已经能够控制他站起来和走路了，我现在就是在解决obevervation的输入问题。

**Comment by Indiasuperman** [comment]
怎么解决的？

**Comment by heartInsert** [comment]
@Indiasuperman  我不知道你用的哪一款机器人，你的SDK版本必须和https://www.yuque.com/ironfatty/nly1un/iktfqz这上面匹配，你不能看github选SDK。

**Comment by Indiasuperman** [comment]
这个版本对的，我匹配过。

**Comment by heartInsert** [comment]
那我就不知道了

**Comment by Indiasuperman** [comment]
@heartInsert 试着把那些privileged info全变0，我现在还因为环境跑不起来

**Comment by Indiasuperman** [comment]
> > 大佬的整体框架没问题我感觉，我准备在这个基础上改observations输入，估计就是输入问题
> > 你的observation 可以了吗，线性速度这个值你有得到吗？

哪个线性速度？

**Comment by heartInsert** [comment]
> > > 大佬的整体框架没问题我感觉，我准备在这个基础上改observations输入，估计就是输入问题
> > > 你的observation 可以了吗，线性速度这个值你有得到吗？
> 
> 哪个线性速度？

我是在isaacgym里面训练的，然后我的observation里面需要机器人的线速度和角速度这2个东西。

**Comment by Indiasuperman** [comment]
> > > > 大佬的整体框架没问题我感觉，我准备在这个基础上改observations输入，估计就是输入问题
> > > > 你的observation 可以了吗，线性速度这个值你有得到吗？
> > 
> > 
> > 哪个线性速度？
> 
> 我是在isaacgym里面训练的，然后我的observation里面需要机器人的线速度和角速度这2个东西。

    def get_obs_buf1(self):
        """
        [self.base_ang_vel  * self.obs_scales.ang_vel]
        躯体角速度          *  0.25(标准化系数）
        世界坐标系角速度？

        :return:
        obs_buf[0:3]
        """
        ang_vel = 0.25
        return torch.Tensor([[self.gyroscope[0], self.gyroscope[1], self.gyroscope[2]]]) * ang_vel 这部分吗

**Comment by heartInsert** [comment]
> > > > > 大佬的整体框架没问题我感觉，我准备在这个基础上改observations输入，估计就是输入问题
> > > > > 你的observation 可以了吗，线性速度这个值你有得到吗？
> > > 
> > > 
> > > 哪个线性速度？
> > 
> > 
> > 我是在isaacgym里面训练的，然后我的observation里面需要机器人的线速度和角速度这2个东西。
> 
> ```
> def get_obs_buf1(self):
>     """
>     [self.base_ang_vel  * self.obs_scales.ang_vel]
>     躯体角速度          *  0.25(标准化系数）
>     世界坐标系角速度？
> 
>     :return:
>     obs_buf[0:3]
>     """
>     ang_vel = 0.25
>     return torch.Tensor([[self.gyroscope[0], self.gyroscope[1], self.gyroscope[2]]]) * ang_vel 这部分吗
> ```
对的，就是这个，IMU雷达里的数据，这个是角速度，但是线速度没有。

**Comment by Indiasuperman** [comment]
@heartInsert 为什么会用到线速度，你的意思是机器人的线速度，而不是每个脚的？

**Comment by Indiasuperman** [comment]
<img width="441" alt="屏幕截图 2024-07-17 210836" src="https://github.com/user-attachments/assets/3669f4e4-d84d-44d6-be02-39b962024b60">
我只找到这个

**Comment by heartInsert** [comment]
每个脚的我也要用，我拿到了，我需要机器人整体的，和12个关节点的。关节点的已经拿到了，整体的在IMU里面，但是差一个线性速度。

**Comment by 1242713693** [comment]
> > 大佬的整体框架没问题我感觉，我准备在这个基础上改观察输入，估计就是输入问题 你的observation 可以了吗，线性速度这个值你有得到吗？

兄弟你好，那个跑酷的代码你部署成功了吗？能看一下你的代码结构吗？谢谢

**Comment by Indiasuperman** [comment]
> > > 大佬的整体框架没问题我感觉，我准备在这个基础上改观察输入，估计就是输入问题 你的observation 可以了吗，线性速度这个值你有得到吗？
> 
> 兄弟你好，那个跑酷的代码你部署成功了吗？能看一下你的代码结构吗？谢谢

我的狗在配环境的时候被我搞坏了，还没部署。我代码结构跟这个开源一样，但是那些privileged info全为0就是我的修改

**Comment by heartInsert** [comment]
> > > 大佬的整体框架没问题我感觉，我准备在这个基础上改观察输入，估计就是输入问题 你的observation 可以了吗，线性速度这个值你有得到吗？
> 
> 兄弟你好，那个跑酷的代码你部署成功了吗？能看一下你的代码结构吗？谢谢

我部署成功了，通过ros收发信息，我可以控制它进行low state的行走了。

**Comment by 1242713693** [comment]
> > > > 大佬的整体框架没问题我感觉，我准备在这个基础上改观察输入，估计就是输入问题 你的observation 可以了吗，线性速度这个值你有得到吗？
> > 
> > 
> > 兄弟你好，那个跑酷的代码你部署成功了吗？能看一下你的代码结构吗？谢谢
> 
> 我部署成功了，通过ros收发信息，我可以控制它进行low state的行走了。

可以参考一下兄弟的代码结构吗？谢谢

**Comment by heartInsert** [comment]
@1242713693  `https://gitee.com/aermang/aliengo_deploy/tree/master/scripts_7_29`  ，我全部放在这里面了，大多数文件都没用，好像就3个py文件有用，程序进入口是real_robot_flow.py , 我用这份代码先在gazebo虚拟环境上验证机器人的行为，然后直接把代码复制到真实机器人上，同样运行real_robot_flow.py  机器人就可以跑了。

**Comment by heartInsert** [comment]
![Uploading gazebo流程.png…]()

**Comment by 1242713693** [comment]
感谢好兄弟


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "XanderWangXY/Deployment-for-Aliengo"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年7月30日(星期二) 下午3:13
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [XanderWangXY/Deployment-for-Aliengo] about deployment (Issue #3)





 
gazebo.png (view on web)

**Comment by Indiasuperman** [comment]
> > > > 大佬的整体框架没问题我感觉，我准备在这个基础上改观察输入，估计就是输入问题 你的observation 可以了吗，线性速度这个值你有得到吗？
> > 
> > 
> > 兄弟你好，那个跑酷的代码你部署成功了吗？能看一下你的代码结构吗？谢谢
> 
> 我部署成功了，通过ros收发信息，我可以控制它进行low state的行走了。

这是跑酷吗？

**Comment by heartInsert** [comment]
@Indiasuperman  就是强化学习的模型，直接控制12个关节点的position，然后行走和奔跑。

**Comment by 1242713693** [comment]
有大佬知道这个问题如
![353822906-358c7dc8-e805-4c16-9282-6875d321370e](https://github.com/user-attachments/assets/17c9305d-9187-4ba6-b3b3-8b971b5e4aa0)
何解决吗？我的系统是Ubuntu20.04 noetic系统

**Comment by heartInsert** [comment]
我也是20.04，但是你这个问题我不知道

**Comment by 1242713693** [comment]
> 我也是20.04，但是你这个问题我不知道

兄弟你好，我按照你的开源的代码在PC端配置环境想去运行，我的运行的环境就是当时跑酷的那个环境“parkour”，发现不断的报错。我想请教一下兄弟，你当时运行的时候是怎样搭建的？我的机器狗是A1，你的这个模型可以用吗？谢谢

**Comment by 1242713693** [comment]
> @Indiasuperman 就是强化学习的模型，直接控制12个关节点的position，然后行走和奔跑。

兄弟你好，我按照你的开源的代码在PC端配置环境想去运行，我的运行的环境就是当时跑酷的那个环境“parkour”，发现不断的报错。我想请教一下兄弟，你当时运行的时候是怎样搭建的？我的机器狗是A1，你的这个模型可以用吗？谢谢

**Comment by 1242713693** [comment]
> 我也是20.04，但是你这个问题我不知道

大佬你好，我可以成功运行你给的.py文件，我同时与运行了unitree_ros中得到gazebo的示例。在.py文件运行后，我查看了ROS发布的主题，可以看到low_cmd和low_state！我想请教一下，如何才能在gazebo中看到机器人的动作呢？希望能够得到大佬的回复，谢谢

**Comment by heartInsert** [comment]
> > 我也是20.04，但是你这个问题我不知道
> 
> 大佬你好，我可以成功运行你给的.py文件，我同时与运行了unitree_ros中得到gazebo的示例。在.py文件运行后，我查看了ROS发布的主题，可以看到low_cmd和low_state！我想请教一下，如何才能在gazebo中看到机器人的动作呢？希望能够得到大佬的回复，谢谢

有个unitree  servo.cpp可以控制运动吧

**Comment by 1242713693** [comment]
感谢大佬的回复
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "XanderWangXY/Deployment-for-Aliengo"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年8月12日(星期一) 下午3:46
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [XanderWangXY/Deployment-for-Aliengo] about deployment (Issue #3)





   
我也是20.04，但是你这个问题我不知道
  
大佬你好，我可以成功运行你给的.py文件，我同时与运行了unitree_ros中得到gazebo的示例。在.py文件运行后，我查看了ROS发布的主题，可以看到low_cmd和low_state！我想请教一下，如何才能在gazebo中看到机器人的动作呢？希望能够得到大佬的回复，谢谢
  
有个unitree  servo.cpp可以控制运动吧

**Comment by heartInsert** [comment]
你看一下real_robot_flow.py的逻辑呗，应该是有个叫public topic的函数去发布了low_cmd_info 这个话题，在那打个断点

**Comment by 1242713693** [comment]
大佬你好，能方便加个QQ或者微信吗？谢谢
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "XanderWangXY/Deployment-for-Aliengo"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年8月12日(星期一) 下午4:49
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [XanderWangXY/Deployment-for-Aliengo] about deployment (Issue #3)





 
你看一下real_robot_flow.py的逻辑呗，应该是有个叫public topic的函数去发布了low_cmd_info 这个话题，在那打个断点

**Comment by heartInsert** [comment]
> 大佬你好，能方便加个QQ或者微信吗？谢谢 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "XanderWangXY/Deployment-for-Aliengo" ***@***.***&gt;; 发送时间:&nbsp;2024年8月12日(星期一) 下午4:49 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [XanderWangXY/Deployment-for-Aliengo] about deployment (Issue #3) 你看一下real_robot_flow.py的逻辑呗，应该是有个叫public topic的函数去发布了low_cmd_info 这个话题，在那打个断点 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: ***@***.***&gt;

361846570
